pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/BaseResource.java

```diff
@@ -324,7 +324,7 @@ public static String getReasonFromServer(WebApplicationException e) {
         } catch (Exception ex) {
             try {
                 return ObjectMapperFactory.getThreadLocal().readValue(
-                        e.getResponse().getEntity().toString(), ErrorData.class).reason;
+                        e.getResponse().readEntity(String.class), ErrorData.class).reason;
```

Comment:
Good catch! could you please add a test for it?

Reply:
Added a unit test, PTAL.